tests/test_responsetypes.py

```diff
@@ -72,17 +72,35 @@ def test_from_headers(self):
             retcls = responsetypes.from_headers(source)
             assert retcls is cls, f"{source} ==> {retcls} != {cls}"
 
-    def test_from_args(self):
-        # TODO: add more tests that check precedence between the different arguments
+    def test_from_args_pre_xtractmime(self):
         mappings = [
             ({'url': 'http://www.example.com/data.csv'}, TextResponse),
-            # headers takes precedence over url
             ({'headers': Headers({'Content-Type': ['text/html; charset=utf-8']}),
-              'url': 'http://www.example.com/item/'}, HtmlResponse),
+              'url': 'http://www.example.com/item/'}, HtmlResponse), # Failing with xtractmime, returning TextResponse expected HtmlResponse
```

Comment:
I figured out it's happening because of step 5 in section 7. Where we implemented essence part as 
```
if supplied_type == b"text/html":
```
and it is not considering `b'text/html; charset=utf-8'` as HTML type. Maybe we can replace the code as
`if supplied_type.startswith(b"text/html")`. Or same can go with the implementation of essence of a mime type in other parts of the xtractmime.

Reply:
Should I open another PR for the above issue in xtractmime?

Reply:
> Should I open another PR for the above issue in xtractmime?

Yes!

Also, try to be careful with the fix. `supplied_type.startswith(b"text/html")` would also include unintended stuff (however unlikely, like `text/htmllikepdfthing`). Ideally, the input MIME types should be parsed with https://mimesniff.spec.whatwg.org/#parsing-a-mime-type . That said, for the current implementation, maybe a simplification that only splits the parameters part out would suffice (e.g. split by `;` and pick the first item).